evennia/utils/validfuncs.py

```diff
@@ -0,0 +1,175 @@
+"""
+Contains all the validation functions.
+
+All validation functions must have a checker (probably a session) and entry arg.
```

Comment:
I think the allowed call signs here should be either `validator(arg)` or `validator(arg, **kwargs)`. Btw - I like that these raise ValueError, makes for a nice and  clean flow - if the validator returns at all, you are golden.

Reply:
My original version of them also receives the Session or Account, so that it could do things such as 'take a datetime string as that specific user favors in their configured timezone, convert it to UTC, and return the UTC version for code purposes.' So I agree with adding something like **kwargs here.

Reply:
Whereas people can add custom validators on whatever form they like, I don't think formulating custom error messages is the work of a validator. It would be better to have the option handler handle the error handling and reporting  - it has all the information available to build a proper, informative error message. The validator could raise a `ValueError' with a custom message string if one wanted to display any further special information about the error.